drivers/samsung/displays/md_series.cr

```diff
@@ -0,0 +1,500 @@
+      end
+    ].flatten
+
+    switch_to(main_source, **options)
+    do_send("screen_split", data, **options)
+  end
+
+  def in_range(val : Int32, max : Int32) : Int32
+    min = 0
+    if val < min
+      val = min
+    elsif val > max
+      val = max
+    end
+    val
+  end
+
+  # Emulate mute
+  def mute_audio(val : Bool = true)
+    if val
+      if !self[:audio_mute]?
+        self[:audio_mute] = true
+        self[:previous_volume] = self[:volume].as_i? || 50
+        volume(0)
+      end
+    else
+      unmute_audio
+    end
+  end
+
+  def unmute_audio
+    if self[:audio_mute]?
+      self[:audio_mute] = false
+      volume(self[:previous_volume].as_i? || 50)
+    end
+  end
+
+  enum SPEAKERMODES
+    Internal = 0
+    External = 1
+  end
+
+  def speaker_select(mode : String, **options)
+    do_send("speaker", SPEAKERMODES.parse(mode).value, **options)
+  end
+
+  def do_poll
+    do_send("status", [] of UInt8, priority: 0)
+    power? unless self[:hard_off]?
+  end
+
+  # Enable power on (without WOL)
+  def network_standby(enable : Bool, **options)
+    state = enable ? 1 : 0
+    do_send("net_standby", state, **options)
+  end
+
+  # Eco auto power off timer
+  def auto_off_timer(enable : Bool, **options)
+    state = enable ? 1 : 0
+    do_send("eco_solution", [0x81, state], **options)
+  end
+
+  # Device auto power control (presumably signal based?)
+  def auto_power(enable : Bool, **options)
+    state = enable ? 1 : 0
+    do_send("auto_power", state, **options)
+  end
+
+  # Display control
+  METHODS = [
+    "volume",
+    "contrast",
+    "brightness",
+    "sharpness",
+    "colour",
+    "tint",
+    "red_gain",
+    "green_gain",
+    "blue_gain"
+  ]
+
+  # Macro to define methods from the array above
+  {% for name in METHODS %}
+    def {{name.id}}(val : Int32, **options)
+      val = in_range(val, 100)
+      do_send({{name.id.stringify}}, val, **options)
+    end
+  {% end %}
+
+  DEVICE_SETTINGS = [
+    :network_standby,
+    :auto_off_timer,
+    :auto_power,
+    :contrast,
+    :brightness,
+    :sharpness,
+    :colour,
+    :tint,
+    :red_gain,
+    :green_gain,
+    :blue_gain,
+]
+
+  def do_device_config
+    logger.debug { "Syncronising device state with settings" }
+    DEVICE_SETTINGS.each do |name|
+      value = setting(Int32, name)
+      # TODO: find out if these are equivalent
+      # __send__(name, value) unless value.nil?
+      do_send(name.to_s, value) unless value.nil?
+    end
+  end
+
+  # TODO: check type for broadcast is correct
+  def wake(broadcast : String? = nil)
+    mac = setting(String, :mac_address)
+    if mac
+      # config is the database model representing this device
+      wake_device(mac, broadcast)
+      info = "Wake on Lan for MAC #{mac}"
+      info += " directed to VLAN #{broadcast}" if broadcast
+      logger.debug { info }
+    else
+      logger.debug { "No MAC address provided" }
+    end
+  end
```

Comment:
Wake on LAN is now provided as part of the [base driver](https://github.com/PlaceOS/driver/blob/60c4604e987fe541cfdbf0367a18f6da385890d2/src/placeos-driver.cr#L145-L147). This can be dropped.